docs/README.md

```diff
@@ -1,21 +1,45 @@
 ## Technical Overview
 
-Kubernetes allows for extensions to its architecture in the form of custom
-resources: <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>.
-KubeVirt represents virtual machines as custom resources and manages changes
-to libvirt domains based on the state of those resources.
+Kubernetes allows for extensions to its architecture via
+[*custom resources*]( https://Kubernetes.io/docs/concepts/extend-Kubernetes/api-extension/custom-resources/),
+which add a new endpoint in the Kubernetes API that stores and retrieves a
+collection API objects of a certain kind.
+The *custom resources* by themselves only
+enable store and retrieve structured data;
+to add business logic and specific functionality,
+[*custom controllers*]( https://Kubernetes.io/docs/concepts/extend-Kubernetes/) are needed.
+*Controllers* are clients of the Kubernetes API-Server that typically read an
+object's `.spec`, possibly do things, and then update the object's
+`.status`.
+
+KubeVirt uses CRDs, *controllers* and other Kubernetes features, to
+represent and manage traditional virtual machines side by side with
+containers.
+
+KubeVirt's primary CRD is the VirtualMachine (VM) resource, which contains
+a VirtualMachineInstance (VMI) object. The VMI
+shares similarity with the Pod concept and represents a single virtualized
+workload that executes once until completion (i.e., powered off). In
+addition to the VMI, the key KubeVirt components are the virt-api, the
+virt-controller, the virt-handler, and the virt-launcher.
```

Comment:
The VMI isn't exactly a component like virt-api, virt-controller, virt-handler, virt-launcher.  The VMI is an API that components interact with.

Reply:
Fixed that!